app/controllers/users/mfa_selection_controller.rb

```diff
@@ -13,16 +13,29 @@ def index
       analytics.user_registration_2fa_additional_setup_visit
     end
 
+    def non_restricted
+      @two_factor_options_form = TwoFactorOptionsForm.new(current_user)
+      @presenter = two_factor_options_presenter
+      render 'index'
+    end
+
     def update
       result = submit_form
       analytics.user_registration_2fa_additional_setup(**result.to_h)
 
       if result.success?
         process_valid_form
+      elsif (result.errors[:selection].include? 'phone') &&
+            IdentityConfig.store.kantara_2fa_phone_restricted
+        flash[:phone_error] = t('errors.two_factor_auth_setup.must_select_additional_option')
+        redirect_to two_factor_options_path(anchor: 'select_phone')
       else
         @presenter = two_factor_options_presenter
         render :index
       end
+    rescue ActionController::ParameterMissing
+      flash[:error] = t('errors.two_factor_auth_setup.must_select_option')
+      redirect_back(fallback_location: two_factor_options_path, allow_other_host: false)
```

Comment:
codeclimate is highlighting this as untested, can we add some test coverage for it?